modules/social_features/social_comment/tests/src/Kernel/CommentViewAccessTest.php

```diff
@@ -0,0 +1,181 @@
+<?php
+
+namespace Drupal\Tests\social_comment\Kernel;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\KernelTests\Core\Entity\EntityKernelTestBase;
+use Drupal\Tests\node\Traits\NodeCreationTrait;
+
+/**
+ * Tests comment view level access.
+ *
+ * @group social_comment
+ */
+class CommentViewAccessTest extends EntityKernelTestBase {
+
+  use NodeCreationTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static $modules = [
+    // For the comment functionality.
+    'social_comment',
+    'comment',
+    // For checking access to comments.
+    'entity',
+    // For the comment author and viewer.
+    'social_user',
+    'user',
+    // User creation in social_user requires a service in role_delegation.
+    "role_delegation",
+    // social_comment configures comments for nodes.
+    'node',
+    // The default comment config contains a body text field.
+    'field',
+    'text',
+    'filter',
+  ];
+
+  /**
+   * The comment storage.
+   *
+   * @var \Drupal\Core\Entity\EntityStorageInterface
+   */
+  private $storage;
+
+  /**
+   * Node entity to use in this test.
+   *
+   * @var \Drupal\node\Entity\Node
+   */
+  private $node;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    $this->installEntitySchema('node');
+    $this->installEntitySchema('user');
+    $this->installEntitySchema('comment');
+    $this->installSchema('comment', 'comment_entity_statistics');
+    $this->installConfig(['filter', 'comment', 'social_comment']);
+
+    $this->storage = $this->entityTypeManager->getStorage('comment');
+    $this->node = $this->createNode();
+  }
+
+  /**
+   * Test that a user can not view their own unpublished comment.
+   *
+   * This mirrors the functionality of the distribution at the time of writing
+   * the test.
+   */
+  public function testUserCanNotViewOwnUnpublishedComment() {
+    // Create an unpublished comment on a node.
+    $user = $this->createUser([], ['access comments']);
+    $this->setCurrentUser($user);
```

Comment:
`setUpCurrentUser` may come in helpful here

```suggestion
    $user = $this->setUpCurrentUser([], ['access comments']);
```